### PR TITLE
5818 get agents by keepalive with chunks

### PR DESCRIPTION
**Pull request by palaciosjeremias**
|Related issue|
|---|
| 5818 |

## Description
This PR adds 
- The query to search for every agent one by one and generate a full list of agents
- The queries to search for agents by keep alive one by one and generate a full list.
- The methods of Wazuh-db that calls this queries and generates chunks of data to be responded.
- The parser for the commands of Wazuh-db for getting this information
- The methods on wdb-agents to create the commands to Wazuh-db and iterate as many times as necessary to obtain a full response.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation

### CODE REVIEW

File: src/wazuh_db/wdb.h [hunk 1]
```diff
@@ -476,9 +481,30 @@ int wdb_create_profile(const char *path);
...
+ * The array is heap allocated memory that must be freed by the caller.
+ * 
+ * @return Pointer to the array, on success.
+ * @retval NULL on errors.
+ */
 int* wdb_get_all_agents();
 
+/**
+ * @brief Returns an array containing the ID of every agent (except 0), ended with -1 based on its keep_alive.
+ * This method creates and sends a command to WazuhDB to receive every the ID of every agent.
```
Comment (pereyra-m) [inline src/wazuh_db/wdb.h:497]: Typo: "every **the** ID"

File: src/wazuh_db/wdb.h [hunk 2]
```diff
@@ -875,6 +923,17 @@ wdb_chunks_status_t wdb_sync_agent_info_get(wdb_t *wdb, int* last_agent_id, char
  */
 int wdb_global_sync_agent_info_set(wdb_t *wdb, cJSON *agent_info);
 
+/**
+ * Doxygen here
```
Comment (pereyra-m) [inline src/wazuh_db/wdb.h:927]: Doxygen description style will be added.

File: src/wazuh_db/wdb.h [hunk 3]
```diff
@@ -875,6 +923,17 @@ wdb_chunks_status_t wdb_sync_agent_info_get(wdb_t *wdb, int* last_agent_id, char
...
+*/
+wdbc_result wdb_global_get_agents_by_keepalive(wdb_t *wdb, int* last_agent_id, char condition, int keep_alive, char **output);
+/**
+ * Doxygen here
```
Comment (pereyra-m) [inline src/wazuh_db/wdb.h:931]: Doxygen description style will be added.

File: src/wazuh_db/wdb_agent.c [hunk 4]
```diff
@@ -437,41 +438,92 @@ int wdb_remove_agent_db(int id, const char * name) {
...
-    char wdboutput[WDBOUTPUT_SIZE] = "";
-
-    sqlite3_snprintf(sizeof(wdbquery), wdbquery,"%s", global_db_queries[SQL_SELECT_AGENTS]);
-    root = wdbc_query_parse_json(&wdb_sock_agent, wdbquery, wdboutput, sizeof(wdboutput));
-
-    if (!root) {
-        merror("Error querying Wazuh DB to get all agents.");
-        return NULL;
+int* wdb_get_agents_by_keepalive(const char* condition, int keepalive) {
+    char wdbquery[OS_BUFFER_SIZE] = "";
```
Comment (pereyra-m) [inline src/wazuh_db/wdb_agent.c:442]: This file has a **WDBQUERY_SIZE** and a **WDBOUTPUT_SIZE** defined.

File: src/wazuh_db/wdb_agent.c [hunk 5]
```diff
@@ -437,41 +438,92 @@ int wdb_remove_agent_db(int id, const char * name) {
...
-    while (item)
-    {
-        json_id = cJSON_GetObjectItemCaseSensitive(item,"id");
-        
-        if(cJSON_IsNumber(json_id)){
-            array[n] = json_id->valueint;
-            n++;
+int* wdb_get_all_agents(void) {
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_MAXSTR] = "";
```
Comment (pereyra-m) [inline src/wazuh_db/wdb_agent.c:488]: This file has a **WDBQUERY_SIZE** and a **WDBOUTPUT_SIZE** defined.

File: src/wazuh_db/wdb_global.c [hunk 6]
```diff
@@ -317,3 +317,170 @@ int wdb_global_sync_agent_info_set(wdb_t *wdb,cJSON * json_agent){
...
+    while (status == WDBC_UNKNOWN) {
+        //Prepare SQL query
+        if (wdb_stmt_cache(wdb, stmt_index) < 0) {
+            mdebug1("cannot cache statement");
+            status = WDBC_ERROR;
+            break;
+        }
+        sqlite3_stmt* stmt = wdb->stmt[stmt_index];
+        if (sqlite3_bind_int(stmt, 1, *last_agent_id) != SQLITE_OK) {
+            merror("DB(%s) sqlite3_bind_text(): %s", wdb->id, sqlite3_errmsg(wdb->db));
```
Comment (pereyra-m) [inline src/wazuh_db/wdb_global.c:355]: Typo: "sqlite3_bind_**text**()..." after a sqlite3_bind_**int**() fail.

File: src/wazuh_db/wdb_global.c [hunk 7]
```diff
@@ -317,3 +317,170 @@ int wdb_global_sync_agent_info_set(wdb_t *wdb,cJSON * json_agent){
...
+            status = WDBC_ERROR;
+            break;
+        }        
+       
+        if (sqlite3_bind_int(stmt, 2, keep_alive) != SQLITE_OK) {
+            merror("DB(%s) sqlite3_bind_text(): %s", wdb->id, sqlite3_errmsg(wdb->db));
```
Comment (pereyra-m) [inline src/wazuh_db/wdb_global.c:361]: Typo: "sqlite3_bind_text()..." after a sqlite3_bind_int() fail.

File: src/wazuh_db/wdb.h [hunk 8]
```diff
@@ -476,9 +481,30 @@ int wdb_create_profile(const char *path);
 /* Create new database file from SQL script */
 int wdb_create_file(const char *path, const char *source);
 
-/* Get an array containing the ID of every agent (except 0), ended with -1 */
+/**
+ * @brief Returns an array containing the ID of every agent (except 0), ended with -1.
+ * This method creates and sends a command to WazuhDB to receive every the ID of every agent.
```
Comment (DProvinciani) [inline src/wazuh_db/wdb.h:486]: Reword this sentence. `to receive every the ID of every agent`

Comment (palaciosjeremias) [reply src/wazuh_db/wdb.h]: Fixed

File: src/wazuh_db/wdb.h (hunk 1 above)
Comment (palaciosjeremias) [reply src/wazuh_db/wdb.h]: Fixed

File: src/wazuh_db/wdb.h (hunk 2 above)
Comment (palaciosjeremias) [reply src/wazuh_db/wdb.h]: Fixed

**Pull request merged**